pkg/asset/agent/manifests/agentpullsecret.go

```diff
@@ -143,43 +146,48 @@ func normalizeDockerConfig(stringData string) (string, error) {
 
 func (a *AgentPullSecret) validatePullSecret() field.ErrorList {
 	if err := a.validateSecretIsNotEmpty(); err != nil {
-		return err
+		return field.ErrorList{err}
 	}
 
-	fieldPath := field.NewPath("StringData")
-	dockerConfig := a.Config.StringData[pullSecretKey]
+	fieldPath := field.NewPath("Data")
+	dockerConfig, ok := a.Config.StringData[pullSecretKey]
+	if !ok {
+		dockerConfig = string(a.Config.Data[pullSecretKey])
+	}
 	if err := validate.ImagePullSecret(dockerConfig); err != nil {
 		return field.ErrorList{field.Invalid(fieldPath, dockerConfig, err.Error())}
 	}
 
 	return field.ErrorList{}
 }
 
-func (a *AgentPullSecret) validateSecretIsNotEmpty() field.ErrorList {
-	var allErrs field.ErrorList
+func (a *AgentPullSecret) validateSecretIsNotEmpty() *field.Error {
+	fieldPath := field.NewPath("Data")
 
-	fieldPath := field.NewPath("StringData")
-
-	if len(a.Config.StringData) == 0 {
-		allErrs = append(allErrs, field.Required(fieldPath, "the pull secret is empty"))
-		return allErrs
+	if len(a.Config.StringData) == 0 && len(a.Config.Data) == 0 {
+		return field.Required(fieldPath, "the pull secret is empty")
 	}
 
 	pullSecret, ok := a.Config.StringData[pullSecretKey]
 	if !ok {
```

Comment:
We should have conflict error if both are present (or maybe just if both are present and they have different values?).